usaspending_api/search/v2/views/spending_by_category_views/awarding_agency.py

```diff
@@ -0,0 +1,89 @@
+from typing import List
+
+from django.db.models import QuerySet, F
+from elasticsearch_dsl import A, Q as ES_Q
+
+from usaspending_api.common.elasticsearch.search_wrappers import TransactionSearch
+from usaspending_api.search.helpers.spending_by_category_helpers import fetch_agency_tier_id_by_agency
+from usaspending_api.search.v2.views.spending_by_category_views.base_spending_by_category import (
+    BaseSpendingByCategoryViewSet,
+    Category,
+)
+
+
+class AwardingAgencyViewSet(BaseSpendingByCategoryViewSet):
+    """
+    This route takes award filters, and returns spending by awarding agencies.
+    """
+
+    endpoint_doc = "usaspending_api/api_contracts/contracts/v2/search/spending_by_category/awarding_agency.md"
+    category = Category(name="awarding_agency", primary_field="awarding_toptier_agency_name.keyword")
+
+    def build_elasticsearch_search_with_aggregations(
+        self, filter_query: ES_Q, curr_partition: int, num_partitions: int, size: int
+    ) -> TransactionSearch:
+        # Create filtered Search object
+        search = TransactionSearch().filter(filter_query)
+
+        # Define all aggregations needed to build the response
+        group_by_agency_name = A(
+            "terms",
+            field="awarding_toptier_agency_name.keyword",
+            include={"partition": curr_partition, "num_partitions": num_partitions},
+            size=size,
+        )
+        group_by_agency_abbreviation = A("terms", field="awarding_toptier_agency_abbreviation.keyword")
+        group_by_agency_id = A("terms", field="awarding_toptier_agency_id")
+        sum_as_cents = A("sum", field="generated_pragmatic_obligation", script={"source": "_value * 100"})
+        sum_as_dollars = A(
+            "bucket_script", buckets_path={"sum_as_cents": "sum_as_cents"}, script="params.sum_as_cents / 100"
+        )
```

Comment:
A lot of this logic is duplicated with awarding subagency. If these two are the only two implementations to use this functionality, I would suggest another abstract class capturing all the things they have in common. If this is more of a quirk and you would expect future implementations to use some but not all of these, I would advise we move these functions to a library. If EVERY implementation will be using all these, we might want to make the base class function not abstract, and have the implementations do the equivilent of a super() call.